### ISSUE TITLE
Use `preview-latex` for sites with MathJax support

### ISSUE THREAD

**Issue by vibhavp** [vermiculus/sx.el#248]
sx.el can AUCTeX's preview-latex functionality (which would require a TeX install) to typset math.

**Comment by vermiculus** [comment]
This is a good idea, so thanks for suggesting it! I bet it would make those sites a lot more intelligible for casual browsers :smile:

This isn't likely to get implemented any time *soon*, seeing as there are broader-impact tasks (#174, #122, #4) that still need work, but this should definitely go in the workings for `v1.0` or later :+1:

**Comment by Bruce-Connor** [comment]
Allow me to quote Sean's comment elsewhere:

> perhaps there is just a function I can run the final buffer text through that will do this

And to add my own idea:

> It might be useful to look into how org does it.

**Comment by vibhavp** [comment]
`org-preview-latex-fragment` uses dvipng to extract math from the file, which it inserts into the buffer. An example of a tex file generated by the function would be:
```latex
\documentclass{article}
\usepackage[usenames]{color}
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{graphicx}
\usepackage[normalem]{ulem}
\usepackage{amsmath}
\usepackage{textcomp}
\usepackage{marvosym}
\usepackage{wasysym}
\usepackage{amssymb}
\tolerance=1000
\pagestyle{empty}
\setlength{\textwidth}{\paperwidth}
\addtolength{\textwidth}{-3cm}
\setlength{\oddsidemargin}{1.5cm}
\addtolength{\oddsidemargin}{-2.54cm}
\setlength{\evensidemargin}{\oddsidemargin}
\setlength{\textheight}{\paperheight}
\addtolength{\textheight}{-\headheight}
\addtolength{\textheight}{-\headsep}
\addtolength{\textheight}{-\footskip}
\addtolength{\textheight}{-3cm}
\setlength{\topmargin}{1.5cm}
\addtolength{\topmargin}{-2.54cm}
\begin{document}
$e^{ix} = \cos(x) + i\sin(x)$
\end{document}

**Comment by TobiasZawada** [comment]
Rendering latex equations in `sx-question-mode` buffers works out of the box with the package `texfrag` from melpa. This ticket is actually a duplicate of the already closed ticket [#331](https://github.com/vermiculus/sx.el/issues/331).

If you want to try `texfrag` it is the easiest way switching on `texfrag-global-mode` and `texfrag-preview-buffer-at-start`. If you want to keep the behavior both options can be customized.

Note that I have used `texfrag` for my personal needs quite a while now. Nevertheless, I recently had to change some stuff for the public release. So it may be that you encounter problems with `texfrag`. You can inform me about those problems by opening an issue at [the official site of `texfrag`](https://github.com/TobiasZawada/texfrag). If those problems are unbearable for you you can just reset the above customization.

Can we also close this ticket now?

**Comment by vermiculus** [comment]
Agreed
